content/articles/2019-08-lifecycle-0-1-0.Rmarkdown

```diff
@@ -0,0 +1,189 @@
+
+The feature is likely to be discontinued in the next major release. Users should switch to an alternative approach as soon as possible.
+
+```{r echo = FALSE}
+knitr::include_graphics("/images/lifecycle/lifecycle-defunct.svg")
+```
+
+The feature can no longer be used. A defunct function is still exported, and a defunct argument is still part of the signature. This way an informative error can be thrown.
+
+You can insert the lifecycle badges anywhere in your documentation. First import the badges in your package:
+
+```{r, eval = FALSE}
+# At the time of writing, you need the development version of usethis
+remotes::install_github("r-lib/usethis")
+usethis::use_lifecycle()
+```
+
+Then use the Rd macro `\lifecycle{stage}`:
+
+```{r, eval = FALSE}
+#' \lifecycle{experimental}
+#' \lifecycle{soft-deprecated}
+```
+
+A good place to include the lifecycle badge of a function is at the top of the `@description` block. To document an argument, the badge can be inserted in the argument description.
+
+Note that you typically don't need to advertise the status if it is the same as the package as a whole. For instance, if your package is [maturing](https://www.tidyverse.org/lifecycle/#maturing), only signal functions in the experimental, stable, questioning stages, or deprecated stages.
+
+
+### Verbosity of deprecation
+
+As is conventional in R, usage of deprecated functions is signalled with warnings. However, lifecycle tries to warn in a non-invasive way. When a deprecated function is called multiple times, only the first time causes a warning. The function then works silently for the rest of the session.
+
+There are two levels of verbosity for deprecated functions, depending on the lifecycle stage. For soft-deprecated features, the verbosity level is minimal. No warning is issued unless the function is called directly from the global environment or from unit tests. This is a way of starting to raise awareness about the deprecation of this function without affecting indirect users.
+
+*   To signal (soft-)deprecation warnings, use `deprecate_soft()` and `deprecate_warn()`:
+
+    ```{r, include = FALSE}
+    # Soft-deprecation warnings are suppressed in the knitr eval env
+    options(lifecycle_verbosity = "warning")
+    ```
+    ```{r}
+    deprecate_soft("0.5.0", "mypkg::foo()")
+    ```
+    ```{r, include = FALSE}
+    options(lifecycle_verbosity = "default")
+    ```
+
+    ```{r}
+    deprecate_warn("0.4.0", "mypkg::bar()")
+    ```
+
+    You can optionally provide a replacement which will be mentioned in the warning message:
+
+    ```{r}
+    deprecate_warn("0.4.0", "mypkg::bar()", "otherpkg::quux()")
+    ```
+
+*   When a function is ready to be effectively discontinued, use `defunct_stop()`:
+
+    ```{r, error = TRUE}
+    deprecate_stop("0.4.0", "mypkg::bar()")
+    ```
+
+*   To control the verbosity of deprecation, use the global option `lifecycle_verbosity`. It can be set to `"quiet"`, `"default"`, `"warning"`, or `"error"`:
```

Comment:
How does this interact with the "warns only once per session" thing?

Reply:
Added:

> Setting the verbosity to `"warn"` forces warnings to occur each time the function is called, it is no longer shown once per session.